### ISSUE TITLE
[CodeGenerator:] Remove ComponentDataProvider without namespace

### ISSUE THREAD

**Issue by sschmid** [sschmid/Entitas-CSharp#323]
Currently, for your convenience, I added a 2nd ComponentDataProvider that ignores namespaces.
I'd like to remove it from the repo and maybe have it as a link in the wiki or get rid of it all together.

**-- Updated because of misunderstandings --**
The code generator should not ignore namespaces if you chose to use them. I personally don't use namespaces for my systems and components. If you decide to use namespaces it will have an impact on the generated api, see comment below.

This will be even more important in combination with #307

**Comment by sschmid** [comment]
Examples
```csharp
namespace MyGame {

    public class HealthComponent : IComponent {
        public int value;
    }
}
```

will generate
```csharp
entity.AddMyGameHealth(42);
```

I personally don't use any namespaces for my game logic

```csharp
public class HealthComponent : IComponent {
    public int value;
}
```
will generate
```csharp
entity.AddHealth(42);
```

Using namespace indicates that there might be name conflicts. That's why the code generator includes the namespace to all generated methods.

**Comment by sschmid** [comment]
While I still think the code generator should not ignore namespaces if you choose to use them, I see a lot of people want to use namespaces because they are "used to use'em" ™️ - that's fair enough. While not officially supported by Entitas you can add this functionality yourself by customizing the component data provider.
In 0.40 I removed `componentName` from the data provider and I'm thinking about readding it for the sake of choice with namespaces. This will allow you to easily use a custom component data provider which set the short type name without namespaces.

In the meantime, you can do this as a temporary solution:
Search for `.Replace("${ComponentName}"` in the folder `Entitas.CodeGenerator/Generators/` and update this

```csharp
//instead of
var componentName = data.GetFullTypeName().ToComponentName();

// do this
var componentName = data.GetFullTypeName().ShortTypeName().RemoveComponentSuffix();
```

**Comment by sschmid** [comment]
I think I found a nice solution. All you have to do is adding this key to your config in order to ignore namespaces

```
Entitas.CodeGeneration.Plugins.IgnoreNamespaces = true
```

**Comment by sschmid** [comment]
Added `IConfigurable`. DataProvider, CodeGenerators and PostProcessors can now be optionally configurable by implementing `IConfigurable`. This means they will need optional keys in `Entitas.properties`, e.g. `Entitas.CodeGeneration.Plugins.IgnoreNamespaces = true`

I also updated the CLI to report if any plugins require keys

```
$ entitas status
Missing key: Entitas.CodeGeneration.Plugins.IgnoreNamespaces
```

This solution will make it easy to customize plugins in the future.
